tests/pixels/test_encoder_base.py

```diff
@@ -450,6 +450,47 @@ def setup_method(self):
         )
         assert self.arr_3s.shape == (4, 2, 3)
 
+    def test_arr_u01_1s(self):
+        """Test processing u1/1s"""
+        self.ds.BitsAllocated = 1
+        self.ds.BitsStored = 1
+        self.ds.SamplesPerPixel = 1
+        self.ds.PixelRepresentation = 0
+        self.ds.Rows = 1
+        self.ds.Columns = 8
+
+        arr = np.asarray([5], dtype="|u1")
+        assert arr.dtype.itemsize == 1
+        self.runner._set_options_ds(self.ds)
+        self.runner.set_source(arr)
+        out = self.runner.get_frame(None)
+
+        # Output data should be bitpacked
+        assert len(out) == 1
+        assert b"\x05" == out
+        assert not self.runner.get_option("is_bitpacked")
+
+    def test_arr_u01_1s_unpacked_source(self):
+        """Test processing u1/1s with unpacked source"""
+        self.ds.BitsAllocated = 1
+        self.ds.BitsStored = 1
+        self.ds.SamplesPerPixel = 1
+        self.ds.PixelRepresentation = 0
+        self.ds.Rows = 1
+        self.ds.Columns = 8
+
+        # The fact that data are not bitpacked should be inferred from the
+        # length
+        arr = np.asarray([1, 0, 1, 0, 0, 0, 0, 0], dtype="|u1")
+        assert arr.dtype.itemsize == 1
+        self.runner._set_options_ds(self.ds)
+        self.runner.set_source(arr)
+        out = self.runner.get_frame(None)
+
+        assert len(out) == 8
+        assert b"\x01\x00\x01\x00\x00\x00\x00\x00" == out
+        assert not self.runner.get_option("is_bitpacked")
```

Comment:
I think this test (and others) need to be explicit in checking the value, as the default will return `None` if `is_bitpacked` isn't found.